hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/spark/sql/hudi/command/IndexCommands.scala

```diff
@@ -90,31 +94,33 @@ case class DropIndexCommand(table: CatalogTable,
   }
 }
 
+/**
+ * Command to show available indexes in hudi. The corresponding logical plan is available at
+ * org.apache.spark.sql.catalyst.plans.logical.ShowIndexes
+ */
 case class ShowIndexesCommand(table: CatalogTable,
                               override val output: Seq[Attribute]) extends IndexBaseCommand {
 
   override def run(sparkSession: SparkSession): Seq[Row] = {
     val metaClient = createHoodieTableMetaClient(table.identifier, sparkSession)
-    val secondaryIndexes = SecondaryIndexManager.getInstance().show(metaClient)
-
-    val mapper = JsonUtils.getObjectMapper
-    toScalaOption(secondaryIndexes).map(x =>
-      x.asScala.map(i => {
-        val colOptions =
-          if (i.getColumns.values().asScala.forall(_.isEmpty)) "" else mapper.writeValueAsString(i.getColumns)
-        val options = if (i.getOptions.isEmpty) "" else mapper.writeValueAsString(i.getOptions)
-        Row(i.getIndexName, i.getColumns.keySet().asScala.mkString(","),
-          i.getIndexType.name().toLowerCase, colOptions, options)
-      }).toSeq).getOrElse(Seq.empty[Row])
+    // need to ensure that the index name is for a valid partition type
+    val indexMetadataOpt = metaClient.getIndexMetadata
```

Comment:
we should first get the indexes available from table config, show those index (excluding files). For secondary and functional index, we can populate the relevant details as you have done here. But, we should not miss record index, partition stats and others if those are available (exept files).

Reply:
Addressed